### ISSUE TITLE
Using `extends` on a non-class-like variable causes a crash, due to the __extends__ helper usage

### ISSUE THREAD

**Issue by hilts-vaughan** [decaffeinate/decaffeinate#1307]
https://decaffeinate-project.org/repl/#?useCS2=false&useJSModules=false&loose=false&evaluate=true&stage=full&code=result%20%3D%20%7B%7D%0Aresult%20extends%20%7B%0A%20desu%3A%20%2212%22%20%0A%7D

If you use this REPL link, you will see what I mean. Yes, apparently that is valid CoffeeScript. If you enter it into a REPL and then log out `result`, you will see it has been extended with a few other properties plus the `desu` property.... the output from decaf crashes. 

yeah, probably insane.

**Comment by alangpierce** [comment]
Got it, yeah, I guess it doesn't surprise me that there might be edge cases here; the `extends` binary operator is a bit obscure and was removed in CS2. Probably the `__extends__` decaffeinate helper can just be tweaked to more closely follow the one used by CoffeeScript.

**Comment by hilts-vaughan** [comment]
Thanks!

I'm just filing issues as I see them from our real live code, I can help patch them as needed but in a lot of cases, the code is just ugly and _bad_ in these edge cases, so I'm rewriting it anyway. :) 

If you don't want certain classifications of problems, please let me know.

**Comment by alangpierce** [comment]
Yeah, my experience has also been that decaffeinate bugs crop up in code that's already buggy or strange-looking. It's probably the right call to rewrite the code in this case, but I think it's still useful for decaffeinate to preserve behavior as much as it can. Bug reports like this are certainly welcome and appreciated!
